swift/ios-shoppe-demo/Views/StoreViewController.swift

```diff
@@ -17,18 +17,20 @@ class StoreViewController: UICollectionViewController, UICollectionViewDelegateF
 
         setupNavigationBar()
 
-        APIService.sharedInstance.getProductsFromFile { (productsRecieved) in
-            if !productsRecieved.isEmpty {
-                self.products = productsRecieved
-                self.products.forEach { (item) in
-                    APIService.sharedInstance.getImageFor(item.imageName) { (image, error) in
-                        if let error = error {
-                            print(error.description)
-                        }
-                        else if let image = image {
-                            item.image = image
-                            DispatchQueue.main.async {
-                                self.collectionView.reloadData()
+        AppStateController.sharedInstance.displayAlert(view: self) {
+            APIService.sharedInstance.getProductsFromFile { (productsRecieved) in
```

Comment:
Do we need to block fetching of products?  The nested code of a products retrieval that does not use the return value from the alert feels unnecessary.